components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2240,7 +2213,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             }
         } else {
             // Anyone who can read a team's information (i.e. any team member) can manage team projects
-            await this.guardTeamOperation(project.teamId, "get");
+            await this.guardTeamOperation(project.teamId || "", "get");
```

Comment:
Not too sure of this behaviour. Previously undefined would result in a `NOT_FOUND` returned by the guard. With my change, it will result `INVALID_ARGUMENT`. If we want, we can catch this behaviour here and convert.

Reply:
Ok with me, I don't think the distinction is that important here/atm. :+1: